life_game/game.rb

```diff
@@ -0,0 +1,151 @@
+class Game
+attr_accessor :board
+SIZE = 30
+
+  def initialize(board)
+    @board = board
+  end
+
+  def self.input_board(input)
+    if input == nil
+      @board = r_board
+    else
+      @board = params_board(input)
+    end
+  end
+
+
+  def self.empty_board
+    Array.new(SIZE){Array.new(SIZE,0)}
+  end
+
+  def self.params_array(input)
+    i=0
+    params_array=[]
+    input.each do |cell|
+      if cell=="1"
+        cell=1
+      elsif input[i+1]=="OFF"||input[i+1]==nil
+          cell=0
+      else
+          cell=cell
+      end
+      params_array<<cell
+      i=i+1
+    end
+    params_array.delete("OFF")
+    params_array
+  end
+
+  def self.params_board(input)
+    params_board = Array.new(SIZE){Array.new}
+    i = 0
+    params_array = params_array(input)
+    params_board.each do |row|
+      params_array[i..i+SIZE-1].each do |cell|
+        row<<cell
+      end
+      i = i+SIZE
+    end
+    params_board
+  end
+
+  def self.r_board
+    top_bottom = Array.new(SIZE*10,0)
+    zeros = Array.new(10,0)
+
+    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
+    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
+    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
+    r4 = zeros+Array.new(6,1)+Array.new(4,0)+zeros
+    r5 = zeros+Array.new(4,1)+Array.new(6,0)+zeros
+
+    top_r = r1+r2+r3+r4+r5
+    bottom_r = zeros+Array.new(3,1)+Array.new(7,0)+zeros
+
+    board = top_bottom+top_r+bottom_r*5+top_bottom
+    start_board = Array.new(SIZE){Array.new}
+    i = 0
+    start_board.each do |row|
+      board[i..i+SIZE-1].each do |cell|
+        row<<cell
+      end
+      i = i+SIZE
+    end
+    start_board
+  end
+
+  def self.neighbours(i,j)
```

Comment:
ta metoda jest bardzo, bardzo trudna :(

przydałoby się jakoś uprościć - na przykład to, że nie potrzebujesz tych przypisań do `neighbours` już by troszkę poprawiło.

Może warto by to rozbić na parę przypadków, które będą zapisane w osobnych metodach?